polkadot/node/network/availability-recovery/src/task.rs

```diff
@@ -222,7 +222,7 @@ impl State {
 		sender
 			.send_message(NetworkBridgeTxMessage::SendRequests(
 				requests,
-				IfDisconnected::TryConnect,
+				IfDisconnected::ImmediateError,
```

Comment:
See https://github.com/paritytech/polkadot/pull/6081

Reply:
thanks! according to what I read on the issue, the cumulus `0002-pov_recovery` test should be failing with the current code of the PR, right? but it's not

I'll roll back to using `TryConnect` just to be safe in this PR.
I'll think about the optimisations we can do in upcoming PRs, because I'd rather have an incremental approach and keep this refactoring backwards-compatible. @eskimor @sandreim sounds good?

If so, can we have this PR merged if it looks good?

Reply:
I investigates this a bit, to me it looks like zombienet omits some of the args we pass to the collator. In the logs, I see no trace of the `--reserved-only` flag that we pass in the toml. https://github.com/paritytech/zombienet/issues/1360

Reply:
Created https://github.com/paritytech/polkadot-sdk/issues/1647 for tracking purposes